agent/logger/audit/audit_log_test.go

```diff
@@ -75,22 +77,55 @@ func TestWritingToAuditLog(t *testing.T) {
 		if len(tokens) != (commonAuditLogEntryFieldCount + getCredentialsEntryFieldCount) {
 
 		}
-		verifyAuditLogEntryResult(logLine, t)
-		verifyCommonAuditLogEntryFieldResult(strings.Join(tokens[:commonAuditLogEntryFieldCount], " "), t)
+		verifyAuditLogEntryResult(logLine, taskArn, t)
+		verifyCommonAuditLogEntryFieldResult(strings.Join(tokens[:commonAuditLogEntryFieldCount], " "), taskArn, t)
 		verifyConstructAuditLogEntryGetCredentialsResult(strings.Join(tokens[commonAuditLogEntryFieldCount:], " "), t)
 	})
 
-	auditLogger.Log(req, dummyResponseCode, GetCredentialsEventType())
+	auditLogger.Log(request.LogRequest{Request: req, Arn: taskArn}, dummyResponseCode, GetCredentialsEventType())
 }
 
-func verifyAuditLogEntryResult(logLine string, t *testing.T) {
-	tokens := strings.Split(logLine, " ")
-	if len(tokens) != (commonAuditLogEntryFieldCount + getCredentialsEntryFieldCount) {
-		t.Fatalf("Incorrect number of tokens in audit log entry. Expected %d.",
-			commonAuditLogEntryFieldCount+getCredentialsEntryFieldCount)
+func TestWritingErrorsToAuditLog(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockInfoLogger := mock_infologger.NewMockInfoLogger(ctrl)
+
+	req, _ := http.NewRequest("GET", "foo", nil)
+	req.RemoteAddr = dummyRemoteAddress
+	parsedUrl, err := url.Parse(dummyUrl)
+	if err != nil {
+		t.Fatal("error parsing dummyUrl")
 	}
-	verifyCommonAuditLogEntryFieldResult(strings.Join(tokens[:commonAuditLogEntryFieldCount], " "), t)
-	verifyConstructAuditLogEntryGetCredentialsResult(strings.Join(tokens[commonAuditLogEntryFieldCount:], " "), t)
+	req.URL = parsedUrl
+	req.Header.Set("User-Agent", dummyUserAgent)
+
+	cfg := &config.Config{
+		Cluster:                 dummyCluster,
+		CredentialsAuditLogFile: "foo.txt",
+	}
+
+	auditLogger := NewAuditLog(dummyContainerInstanceArn, cfg, mockInfoLogger)
+
+	if auditLogger.GetCluster() != dummyCluster {
+		t.Fatal("Cluster is not initialized properly")
+	}
+
+	if auditLogger.GetContainerInstanceArn() != dummyContainerInstanceArn {
+		t.Fatal("ContainerInstanceArn is not initialized properly")
+	}
+
+	mockInfoLogger.EXPECT().Info(gomock.Any()).Do(func(logLine string) {
+		tokens := strings.Split(logLine, " ")
+		if len(tokens) != (commonAuditLogEntryFieldCount + getCredentialsEntryFieldCount) {
```

Comment:
This will be verified in verifyAuditLogEntryResult, do we need to check here? Same happened above line: 77.